framework/src/modules/nft/schemas.ts

```diff
@@ -280,12 +280,18 @@ export const getCollectionIDsRequestSchema = {
 			minLength: LENGTH_CHAIN_ID * 2,
 			maxLength: LENGTH_CHAIN_ID * 2,
 		},
+		collectionID: {
+			type: 'string',
+			format: 'hex',
+			minLength: LENGTH_COLLECTION_ID * 2,
+			maxLength: LENGTH_COLLECTION_ID * 2,
+		},
```

Comment:
Why does it need `collectionID` as an input ?

Reply:
For validation in the `getSupportedCollectionIDs` endpoint?

```ts
validator.validate<{ chainID: string; collectionID: string }>(
	getSupportedCollectionIDsRequestSchema,
	params,
);

Reply:
> The collectionID is used to get the nftID
you could simply pad zeroes to get the nftID. Collection id is not needed for this endpoint.

Reply:
By padding with zeros, 

```ts
if (supportForNftChainIdExists) {
	const supportedNFTsStoreData = await supportedNFTsStore.get(methodContext, nftChainID);
	if (supportedNFTsStoreData.supportedCollectionIDArray.length === 0) {
		return true;
	}

	const collectionID = this.getCollectionID(nftID);
	if (
		supportedNFTsStoreData.supportedCollectionIDArray.some(id =>
			collectionID.equals(id.collectionID),
		)
	) {
		return true;
	}
}

return false;
```

will always return false when you pass supported collection(s). 

When `supportedNFTsStoreData` has data, `supportedNFTsStoreData.supportedCollectionIDArray.length` will be 
`> 0` hence the check `if (supportedNFTsStoreData.supportedCollectionIDArray.length === 0)` will be skipped.

`const collectionID = this.getCollectionID(nftID);` will be a zero buffer due to the zero padding and as such 
`collectionID.equals(id.collectionID)` will always be false.

Reply:
A couple of inquiries, since we need to return supported collections against the provided chain, would if suffice if the key (chainID) exists in the store and perhaps the check if the chain is supported or not is not required? 🤔  If chainID (key) doesn't exist in the store return `{ collectionIDs: undefined }` and the rest of the logic could remain same? Could something like this be considered sensible?

```ts
	public async getSupportedCollectionIDs(
		context: ModuleEndpointContext,
	): Promise<{ collectionIDs: string[] | undefined }> {
		const { params } = context;

		validator.validate<{ chainID: string }>(
			getSupportedCollectionIDsRequestSchema,
			params,
		);

		const chainID = Buffer.from(params.chainID, 'hex');

		const supportedNFTsStore = this.stores.get(SupportedNFTsStore);

		const chainExists = await supportedNFTsStore.has(context.getImmutableMethodContext(), chainID);

		if (!chainExists) {
		// undefined because empty collectionIDs as per LIP, means all collections are supported!?!
			return { collectionIDs: undefined };
		}

		const supportedNFTsData = await supportedNFTsStore.get(
			context.getImmutableMethodContext(),
			chainID,
		);

		if (supportedNFTsData.supportedCollectionIDArray.length) {
			return {
				collectionIDs: supportedNFTsData.supportedCollectionIDArray.map(collection =>
					collection.collectionID.toString('hex'),
				),
			};
		}

		return { collectionIDs: [] };
	}
```

Reply:
hmm 🤔....nice take on this. Thinking about it now as well....😎

Reply:
@mosmartin @has5aan The logic should be in line with this:

1. Check if there is an entry with key `ALL_SUPPORTED_NFTS_KEY`. If yes, this means all collections for the provided `chainID` are supported (although the store doesn't store them explicitly).

2. If the first check does not pass, check if there is an entry with key equal to `chainID`. 
- If no entry is found, this means no `collectionID` from that chain is supported. 
- If an entry is found, proceed to check the `supportedCollectionIDArray` associated with the `chainID` key. If `supportedCollectionIDArray` = [], this means all collections for that `chainID` are supported (although, again, the store doesn't store them explicitly). If `supportedCollectionIDArray` is non-empty, it lists the `collectionID`s of the supported collections for that `chainID`, so return `supportedCollectionIDArray`.

Hope this helps, if you need me I can write it in the pseudocode.

P.S. @Incede Yes, there is no need for `collectionID` as input.

Reply:
@mjerkov thanks, valid point, the check for ALL_SUPPORTED_NFTS_KEY is missing here, thanks for the correction. :pray:

Reply:
As far as I'm concerned, this can be resolved.